### PR TITLE
feat(instance): add more tests and fix them

### PR DESCRIPTION
**Pull request by kindermoumoute**
feat: shutdown option on delete server

### CODE REVIEW

File: internal/namespaces/instance/v1/testdata/test-create-server-simple-gp1xs.stdout.golden [hunk 1]
```diff
@@ -0,0 +1,35 @@
+id                        f018721d-0f0b-4229-a0a7-ef00cb91d1e8
+name                      cli-srv-intelligent-euler
+organization              b2593aa3-d0e8-4366-89c3-6e666abe1f6f
+allowed-actions.0         poweron
+allowed-actions.1         backup
+commercial-type           GP1-XS
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/testdata/test-create-server-simple-gp1xs.stdout.golden]: we should use `testing.go/TestCheckEqual()` to only test for this value

`.golden` should be used for UI testing 

see https://github.com/scaleway/scaleway-cli/blob/v2/internal/namespaces/instance/v1/instance_cli_test.go#L153

File: internal/namespaces/instance/v1/testdata/test-create-server-simple-with-bootscript.stdout.golden [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+image.public              true
+image.root-volume         dd5f5c10-23b1-4c9c-8445-eb6740957c84
+image.state               available
+protected                 false
+public-ip.id              4ff500c6-163a-4079-aff7-923bc2dee5db
+public-ip.address         51.158.103.52
+public-ip.dynamic         false
+modification-date         few seconds ago
+state                     archived
+bootscript                x86_64 longterm 4.14 latest
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/testdata/test-create-server-simple-with-bootscript.stdout.golden]: we should use `testing.go/TestCheckEqual()` to only test for this value

File: internal/namespaces/instance/v1/testdata/test-create-server-simple-with-name.stdout.golden [hunk 3]
```diff
@@ -0,0 +1,35 @@
+id                        6427d42e-e84f-4db0-bd6f-952dfc3171c4
+name                      yo
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/testdata/test-create-server-simple-with-name.stdout.golden]: we should use `testing.go/TestCheckEqual()` to only test for this value

File: internal/namespaces/instance/v1/testdata/test-create-server-simple-with-start.stdout.golden [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+image.public              true
+image.root-volume         dd5f5c10-23b1-4c9c-8445-eb6740957c84
+image.state               available
+protected                 false
+private-ip                10.68.202.41
+public-ip.id              902dae7f-302d-4be3-a821-f361b652f308
+public-ip.address         51.15.134.93
+public-ip.dynamic         false
+modification-date         few seconds ago
+state                     running
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/testdata/test-create-server-simple-with-start.stdout.golden]: we should use `testing.go/TestCheckEqual()` to only test for this value

File: internal/namespaces/instance/v1/instance_cli_test.go [hunk 5]
```diff
@@ -4,12 +4,19 @@ import (
 	"testing"
 
 	"github.com/scaleway/scaleway-cli/internal/core"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/stretchr/testify/assert"
 )
 
+func init() {
+	if !core.UpdateGolden {
+		instance.RetryInterval = 0
```
Comment (loicbourgois) [inline internal/namespaces/instance/v1/instance_cli_test.go]: why not do it next to `var UpdateGolden = os.Getenv("UPDATE_GOLDEN") == "true"` ?

File: internal/namespaces/instance/v1/custom_server.go [hunk 6]
```diff
@@ -293,6 +294,10 @@ func serverDeleteCommand() *core.Command {
 				Name:  "delete-volumes",
 				Short: "Delete the volumes attached to the server as well",
 			},
+			{
+				Name:  "force-shutdown",
+				Short: "Delete the server even if it is running",
```
[the remark this answers is not in the document]
Comment (kindermoumoute) [reply internal/namespaces/instance/v1/custom_server.go]: This arg spec is in the `scw instance server delete` command

File: internal/namespaces/instance/v1/instance_cli_test.go (hunk 5 above)
Comment (kindermoumoute) [reply internal/namespaces/instance/v1/instance_cli_test.go]: We scope it per namespace in the SDK, I just put it there to be consistent

**Pull request merged**